src/shared/components/VideoPreviewBase/VideoPreviewBase.tsx

```diff
@@ -37,6 +48,23 @@ export type VideoPreviewBaseMetaProps = {
   onChannelClick?: (e: React.MouseEvent<HTMLElement>) => void
   onCoverResize?: (width: number | undefined, height: number | undefined) => void
 }
+
+type VideoPreviewPublisherProps =
+  | {
+      publisherMode: true
+      videoPublishState?: 'default' | 'draft' | 'unlisted'
+      selected: boolean
+      isAnyVideoSelected: boolean
+      onSelectClick: (value: boolean) => void
+    }
+  | {
+      publisherMode?: false | undefined
+      videoPublishState?: undefined | 'default'
+      selected?: undefined
+      isAnyVideoSelected?: undefined
+      onSelectClick?: undefined
+    }
+
```

Comment:
Used to avoid impossible states that might lead to bugs i.e. `publisherMode=false` while at the same time `videoPublishState='draft'`